presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -332,15 +334,51 @@ public ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSessi
     @Override
     public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName)
     {
-        if (isPartitionsSystemTable(tableName)) {
-            return getPartitionsSystemTable(session, tableName);
+        if (PARTITIONS_TABLE_HANDLER.matches(tableName)) {
+            return getPartitionsSystemTable(session, tableName, PARTITIONS_TABLE_HANDLER.getSourceTableName(tableName));
+        }
+        if (PROPERTIES_TABLE_HANDLER.matches(tableName)) {
+            return getPropertiesSystemTable(session, tableName, PROPERTIES_TABLE_HANDLER.getSourceTableName(tableName));
         }
         return Optional.empty();
     }
 
-    private Optional<SystemTable> getPartitionsSystemTable(ConnectorSession session, SchemaTableName tableName)
+    private Optional<SystemTable> getPropertiesSystemTable(ConnectorSession connectorSession, SchemaTableName tableName, SchemaTableName sourceTableName)
+    {
+        Optional<Table> table = metastore.getTable(sourceTableName.getSchemaName(), sourceTableName.getTableName());
+        if (!table.isPresent() || table.get().getTableType().equals(TableType.VIRTUAL_VIEW.name())) {
+            throw new TableNotFoundException(tableName);
+        }
+        List<Map.Entry<String, String>> sortedEntries = table.get().getParameters().entrySet().stream().sorted(Comparator.comparing(Map.Entry::getKey)).collect(toImmutableList());
+        List<ColumnMetadata> columns = sortedEntries.stream().map(Map.Entry::getKey).map(key -> new ColumnMetadata(key, VarcharType.VARCHAR)).collect(toImmutableList());
+        List<Type> types = columns.stream().map(ColumnMetadata::getType).collect(toImmutableList());
+        Iterable<List<Object>> propertyValues = ImmutableList.of(ImmutableList.copyOf(sortedEntries.stream().map(Map.Entry::getValue).collect(toImmutableList())));
```

Comment:
This could be simplified to
```java
Map<String, String> sortedTableParameters = ImmutableSortedMap.copyOf(table.get().getParameters());
List<ColumnMetadata> columns = sortedTableParameters.entrySet().stream()
```
Use a sorted map rather than `List<Entry>`